app/controllers/registrations_controller.rb

```diff
@@ -463,7 +455,8 @@ def payment_completion
     # Some API gateways like Stripe provide the client_secret as a kind of "checksum" (or fraud protection)
     #   back to us upon redirect. Other providers (like PayPal…) unfortunately don't.
     #   So we only compare this secret value with our stored intent record if it's actually provided to us
-    if secret_check.present? && stored_intent.client_secret != secret_check
+    validation_token = params[:payment_intent_client_secret]
+    if validation_token.present? && stored_intent.client_secret != validation_token
```

Comment:
I think `validation_token` is a bit more general and easier to grok than `client_secret` in this context - we could rename the column, or have a `validate_token` method in PaymentIntent - but I didn't want to be too sweeping in the changes, I think the rename already brings a lot of clarity

Reply:
Done